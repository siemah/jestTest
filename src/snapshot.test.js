/**
 * snapshots are made to test some behaviors
 * who change most of the times there is 2 kind:
 * 1- snapshot with toMatchSnapshot() method
 * 2- inline snapshot using toMatchInlineSnapshot()
 * Best practice:
 *  - Treat snapshot as code, this mean review of snapshot like code review
 *  - Tests should be deterministic, use mock function for more consistancy 
 *  about return values who keep changing every time like dates, timers, responses ..
 *  - Use descriptive snapshot names
 */
describe('Snapshots', () => {
  let user;
  beforeEach( done => {
    user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
      name: 'Sven bender',
    };
    done();
  })

  it('will fail every time', () => {
    expect(user).toMatchSnapshot();
  });

  it('should use matcher prop to expect the object props', () => {
    expect(user).toMatchSnapshot({
      createdAt: expect.any(Date),
      id: expect.any(Number),
      name: expect.any(String)
    })
  })

})
