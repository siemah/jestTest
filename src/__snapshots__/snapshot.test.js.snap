// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshots should use matcher prop to expect the object props 1`] = `
Object {
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": Any<String>,
}
`;

exports[`Snapshots will fail every time 1`] = `
Object {
  "createdAt": 2019-11-01T16:09:01.677Z,
  "id": 4,
  "name": "LeBron James junior",
}
`;
